source/shadow.c

```diff
@@ -297,6 +398,41 @@ static ShadowStatus_t validateName( const char * pString,
 
 /*-----------------------------------------------------------*/
 
+static ShadowStatus_t extractThingName( const char * pTopic,
+                                        uint16_t topicLength,
+                                        uint16_t * pConsumedTopicLength,
+                                        uint16_t * pThingNameLength )
+{
+    ShadowStatus_t shadowStatus;
+
+    /* Extract thing name. */
+    shadowStatus = validateName( &( pTopic[ *pConsumedTopicLength ] ),
+                                 topicLength - *pConsumedTopicLength,
+                                 pThingNameLength );
+
+    if( shadowStatus != SHADOW_SUCCESS )
+    {
+        shadowStatus = SHADOW_THINGNAME_PARSE_FAILED;
+        LogDebug( ( "Not related to Shadow. Failed to parse thing name in pTopic %s.", pTopic ) );
+    }
+    else if( *pThingNameLength > SHADOW_THINGNAME_MAX_LENGTH )
```

Comment:
Move this check within `validateName()` (lines 385/386).  It doesn't make sense that `validateName()` returns `SHADOW_SUCCESS` if the thing name is too long.  This will also simplify this function.

Reply:
Yea. That is a good point. Thanks